### PR TITLE
Update Account Subtype Typing Using Classes

### PR DESCRIPTION
**Pull request by zsweigart**
Classes will allow us to have a strongly typed system that matches Android & iOS.  This also updates the parsing of the json in the Android native module.

### CODE REVIEW

**Comment by pladaub** [comment]
I like this API. It makes it impossible for trivially invalid Account Subtypes to be constructed, and matches the pattern used in the Objective-C bridge.

One feature compatibility point to call out: Objective-C allows for unknown Account Subtypes to be constructed by the client developer. This may be something to consider here, in case an Account Subtype is added and requires integration support before SDK support lands.

Aside from this, LGTM. Happy to approve pending alignment with Jeroen.

File: Types.ts
Comment (JeroenMols) [inline Types.ts:23]: Should we rename this to `LinkAccountSubtype`?

```suggestion
    accountSubtypes?: Array<LinkAccountSubtype>;
```

This is part of our public interface and for our customers this is what it actually represents.

Since `LinkAccountType` and `LinkAccountSubType` are private to our javascript module, we can name them either `AccountType` or `LinkAccountTypeEnum`.

File: android/src/main/java/com/plaid/PlaidModule.kt
Comment (JeroenMols) [inline android/src/main/java/com/plaid/PlaidModule.kt:146]: 💅

```suggestion
       val subtypeList = subtypesArray.map {
        val subtypeObject = subtypesArray.get(i) as JSONObject
        LinkAccountSubtype.convert(subtypeObject.getString(SUBTYPE), subtypeObject.getString(TYPE))
      }
```

**Review by JeroenMols — APPROVED** [review]
I do feel we should rename the `LinkAccountSubtypeInterface`

**Comment by zsweigart** [comment]
@pladaub @JeroenMols on Android we have `OTHER` for inputs and `UNKNOWN` for outputs that are defined, I added `LinkAccountSubtypeOther` here to handle the input side of things.  The output side isn't really well-handled in typescript but because it's javascript it won't break anything.

**Comment by pladaub** [comment]
> @pladaub @JeroenMols on Android we have `OTHER` for inputs and `UNKNOWN` for outputs that are defined, I added `LinkAccountSubtypeOther` here to handle the input side of things. The output side isn't really well-handled in typescript but because it's javascript it won't break anything.

This is a divergence from how iOS handles things, it sounds like there are inconsistencies in how iOS and Android represent account types.

Let's discuss internally, the issues already exist and so don't block this PR.

**Review by pladaub — APPROVED** [review]

**Pull request merged**
